Add process metadata file for work with eBPF agent (#156)
SkyAPM/go2sky: README.md

|:-----------------:|:----------------------------------------------------------------------------------------:|:--------------------:|
| agent.sample_rate | The percentage of trace when sampling. It's `[0, 1]`, Same with `WithSampler` parameter. |         0.1          |

> ## Process

Reviewer: ```suggestion
## Process Status Hook
```

Process is a common and wide used concept in Linux, I don't want to hijack it as a feature name.

Author: Updated.

> This feature is used in cooperation with the [skywalking-rover](https://github.com/apache/skywalking-rover) project.

Reviewer: I think you missed one, I suggest this feature(temp files) should be optional and OFF on default.

Reviewer 2: It is recommended to enable this feature through environment variable

Author: Updated and let the process label and process status hook enabled could update through the environment variable.

When go2sky keeps alive with the backend, it would write a metadata file to the local (temporary directory) at the same time, which describes the information of the current process.
The rover side scans all processes, find out which process contains this metadata file, and update the confirmed file. Finally, the rover could collect, profiling, with this process.

### Metadata File

> The metadata file use to save metadata with current process, it save in: `{TMPDIR}/apache_skyuwalking/process/{pid}/metadata.properties`.

Author: Only the doc typo, updated.

Also, when the go2sky keep alive with backend, modify and open time of the metadata file would be updated.

| Key | Type | Description |
|-----|------|------------|
|layer|string|this process layer.|
|service_name|string|this process service name.|
|instance_name|string|this process instance name.|
|process_name|string|this process process name, it's same with the instance name.|
|properties|json|the properties in instance, the process labels also in the properties value.|

> |label_key_in_properties|string|the key name of the process labels.|

Reviewer: I think don't need to be so complex, how about putting labels in the file directly?

Author: Update to the as the `labels` property key, and write the label values directly.

|language|string|current process language, which is `golang`.|

> ### Confirmation file

Reviewer: ```suggestion
### ACK file
```

Reviewer: Please change file and description accordingly. Confirmation feels strange.

> The confirmation file used to let the go2sky known, the rover side have detected current process, so don't need to keep update the metadata file again. It save in: `{TMPDIR}/apache_skyuwalking/process/{pid}/metadata-confirm.properties`.

Reviewer: ```suggestion
The confirmation file is to let the go2sky know, the rover side has detected the current process, so don't need to keep updating the metadata file again. It save in: `{TMPDIR}/apache_skywalking/process/{pid}/metadata-confirm.properties`.
```

Same issue here.

Also, I am not sure this file `doesn't need to keep udpate`? Then what happens if rover crashes or exits?

Author: > Then what happens if rover crashes or exits?

Yes, that's a problem. In this way, maybe we don't need to have the ack mechanism. because it useless to go2sky? WDYT?

Reviewer: Agree, that increases the complexity.

Author: Deleted the ack mechanism.

|:-----------------:|:----------------------------------------------------------------------------------------:|:--------------------:|
| agent.sample_rate | The percentage of trace when sampling. It's `[0, 1]`, Same with `WithSampler` parameter. |         0.1          |

## Process Status Hook

This feature is used in cooperation with the [skywalking-rover](https://github.com/apache/skywalking-rover) project.

> When go2sky keeps alive with the backend, it would write a metadata file to the local (temporary directory) at the same time, which describes the information of the current process.
> The rover side scans all processes, find out which process contains this metadata file. Finally, the rover could collect, profiling, with this process.

Reviewer: ```suggestion
When go2sky keeps alive with the backend, it would write a metadata file to the local (temporary directory) at the same time, which provides the information of the current process.
The rover would detect this file in order to identify this Golang service, and tags it `ebpf-profiling-able`.
```

As a doc, we don't need to describe how codes work, they are open sourced :) We need to tell why it works in this way.

Author: `ebpf-profiling-able`? Do you mean to add this tag to the process labels?

Reviewer: I mean logically, this is what this is designed for. Nothing about label.
